frontend/components/App.js

```diff
@@ -1,10 +1,116 @@
 import React from 'react'
 
+import Form from './Form'
+import TodoList from './TodoList'
+
+const todos = [{
+  name: 'Do dishes',
+  id: 0,
+  completed: false
+}, 
+{
+  name: 'Clean room',
+  id: 1, 
+  completed: false
+}]
+
 export default class App extends React.Component {
+  constructor(){
+    super()
+    this.state = {
+      todos: todos,
+      input: ''
+    }
+  }
+
+  //TodoList Component Handlers:
+  //Crossing out an item once clicked on
+  toggleTodo = (id) => {
+   const newTodoList = this.state.todos.map(item => {
+    if(item.id === id){
+      return {
+        ...item, completed: !item.completed
+      }
+    }
+    else{
+      return(item)
+    }
+   })
+
+   this.setState({
+    todos: newTodoList
+   })
+  }
+
+  //Form Component Handlers: 
+  //Adding an item to the list:
+  addTodo = (item) => {
+    const newTodo = {
+      name: item,
+      id: this.state.todos.length, 
+      completed: false
+    }
+
+    const newTodos = [...this.state.todos, newTodo]
+
+    this.setState({
+      todos: newTodos
```

Comment:
Since the variable newTodos isn't getting changed you can use
`this.setState({
    todos: [...this.state.todos, newTodo]
})`
instead